tensorboard/plugins/interactive_inference/witwidget/notebook/colab/wit.py

```diff
@@ -233,9 +236,14 @@ def set_examples(self, examples):
       self._generate_sprite()
 
   def infer(self):
-    inferences = base.WitWidgetBase.infer_impl(self)
-    output.eval_js("""inferenceCallback('{inferences}')""".format(
-      inferences=json.dumps(inferences)))
+    try:
+      inferences = base.WitWidgetBase.infer_impl(self)
+      output.eval_js("""inferenceCallback('{inferences}')""".format(
```

Comment:
Pardon the drive-by, but this is an injection vector when any of the
keys or values (recursively) of `inferences` contains a single quote:

```python
>>> import json
>>> inferences = {"label_vocab": "',alert('hmm'),null[0],'"}
>>> js = """inferenceCallback('{inferences}')""".format(
...     inferences=json.dumps(inferences))
>>> print(js)
inferenceCallback('{"label_vocab": "',alert('hmm'),null[0],'"}')
```

I’d expect to see simply `inferenceCallback({inferences})`, without the
quotes, such that the JS gets an object literal rather than a string
literal. If it really needs to be a string for some reason, you can
create the string literal safely by JSON-encoding again (in either
Python or JS).

Reply:
replaced with object and not string.